Core/MIPS/IR/IRJit.cpp

```diff
@@ -210,7 +210,55 @@ void IRBlockCache::RestoreSavedEmuHackOps(std::vector<u32> saved) {
 	}
 }
 
-bool IRBlock::HasOriginalFirstOp() {
+JitBlockDebugInfo IRBlockCache::GetBlockDebugInfo(int blockNum) const {
+	const IRBlock &ir = blocks_[blockNum];
+	JitBlockDebugInfo debugInfo{};
+	uint32_t start, size;
+	ir.GetRange(start, size);
+	debugInfo.originalAddress = start;  // TODO
+
+	for (u32 addr = start; addr < start + size; addr += 4) {
+		char temp[256];
+		MIPSDisAsm(Memory::Read_Instruction(addr), addr, temp, true);
+		std::string mipsDis = temp;
+		debugInfo.origDisasm.push_back(mipsDis);
+	}
+
+	for (int i = 0; i < ir.GetNumInstructions(); i++) {
+		IRInst inst = ir.GetInstructions()[i];
+		char buffer[256];
+		DisassembleIR(buffer, sizeof(buffer), inst);
+		debugInfo.irDisasm.push_back(buffer);
+	}
+	return debugInfo;
+}
+
+void IRBlockCache::ComputeStats(BlockCacheStats &bcStats) const {
+	// TODO: Implement properly
+	memset(&bcStats, 0, sizeof(bcStats));
+	bcStats.numBlocks = (int)blocks_.size();
+}
+
+int IRBlockCache::GetBlockNumberFromStartAddress(u32 em_address, bool realBlocksOnly) const {
+	u32 startPage = AddressToPage(em_address);
+	u32 endPage = AddressToPage(em_address + 4);
```

Comment:
Hmm, do we ever need to check the end address?  I think we only need the em_address.

-[Unknown]

Reply:
Not really, I was just lazy copy-pasting code from another function, which might indicate there's room for a utility :) I'll clean it up shortly.